pgmpy/utils/mathext.py

```diff
@@ -179,22 +180,58 @@ def sample_discrete_maps(states, weight_indices, index_to_weight, size=1, seed=N
     if seed is not None:
         np.random.seed(seed)
 
-    # TODO: Remove this conversion and find a way to do this natively in torch.
-    states = np.array(states)
-    weight_indices = compat_fns.to_numpy(weight_indices)
-    index_to_weight = {
-        key: compat_fns.to_numpy(value) for key, value in index_to_weight.items()
-    }
-    size = int(size)
+    # Handle PyTorch tensors natively if we're using the torch backend
+    if config.get_backend() == "torch" and isinstance(weight_indices, torch.Tensor):
```

Comment:
@Akhilesh-max The idea is not to do this if-else block for torch and numpy. Ideally, we would like to handle both in a single block without such if-else logic. To make some of the functions agnostic to the array type we implement functions in `compat_fns.py` that applies the relevant operation depending on whether the array is numpy or torch.